### PR TITLE
✨ feat(command palette): support "table" SearchInput mode

### PR DESCRIPTION
**Pull request by tnyo43**
## Issue

~- resolve:~

## Why is this change needed?

To support column search in the CommandPalette, we would like to introduce a new input mode which filters options with specific tables.
This PR will introduce "table" input mode with a table name. It is not activatable by default, so that users will not see for a while.

### demo

I created a preview with table mode activatable -> https://liam-erd-sample-7mknrlgc9-liambx.vercel.app
You can press "Tab" key when selecting a table option to go into the table mode with it. We haven't implemented the table mode behaviours, so it will be empty, but you can see the input element changing.

https://github.com/user-attachments/assets/ca89d009-6d23-48fa-b32d-035f1d8eaf12

## Summary by CodeRabbit

* **New Features**
  * Command palette shows contextual suggestions (table or command) and can enter a table/column mode that displays "tableName /" when accepted.
  * Tab (when enabled) can accept a table suggestion and switch to table mode, clearing the input; '>' still triggers command mode.

* **Tests**
  * Added tests for suggestion-driven behavior, Tab acceptance, Backspace transitions, and related mode edge cases.

### CODE REVIEW

File: frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteContent/CommandPaletteContent.test.tsx [hunk 1]
```diff
@@ -157,6 +157,23 @@ describe('input mode and filters', () => {
...
+      await user.keyboard('{Tab}')
+
+      // command options still be displayed because table mode is not activated
+      expect(
+        screen.getByRole('option', { name: 'Copy Link ⌘ C' }),
+      ).toBeInTheDocument()
+      expect(
+        screen.getByRole('option', { name: 'Zoom to Fit ⇧ 1' }),
+      ).toBeInTheDocument()
+    })
```
Comment (tnyo43) [inline frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteContent/CommandPaletteContent.test.tsx:176]: I checked this test will fail when I add `isTableModeActivatable={true}` as a prop of [`CommandPaletteSearchInput` in `CommandPaletteContent`](https://github.com/liam-hq/liam/blob/05362c87a81f3c708865f99318c4568f29574d5e/frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteContent/CommandPaletteContent.tsx#L43-L48) which mean the flag works as expected.

File: frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx [hunk 2]
```diff
@@ -28,38 +38,52 @@ export const CommandPaletteSearchInput: FC<Props> = ({
...
+          // TODO: remove this condition and always activate table mode when releasing the feature
+          if (isTableModeActivatable) {
+            if (event.key === 'Tab' && suggestion?.type === 'table') {
+              // switch to "table" mode if a table is suggested and Tab key is pressed
+              event.preventDefault()
+              setMode({ type: 'table', tableName: suggestion.name })
+              setValue('')
+              return
+            }
+          }
```
Comment (tnyo43) [inline frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx:67]: It doesn't turn to table mode yet since `isTableModeActivatable` is `false` in the release.
I added a test case in `CommandPaletteContent` to make sure the `Tab` doesn't work.

File: frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx
Comment (tnyo43) [inline frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx:1]: I did some refactoring at the same time 🙇 
Please check the code hiding whitespaces.

File: frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx [hunk 3]
```diff
@@ -72,6 +96,8 @@ export const CommandPaletteSearchInput: FC<Props> = ({
           onValueChange={setValue}
           className={styles.input}
           placeholder="Search"
+          autoFocus
+          onKeyDown={handleKeydown}
```
Comment (tnyo43) [inline frontend/packages/erd-core/src/features/erd/components/ERDRenderer/CommandPalette/CommandPaletteSearchInput/CommandPaletteSearchInput.tsx:100]: I thought adding event listeners with `useEffect` is more complicated and more dengerous, so I use `onKeyDown` instead.
`autoFocus` is not necessary in the application since cmdk originally support the `autoFocus`. However it's required to replicate the same behavior in tests.

**Review by junkisai — APPROVED** [review]
LGTM!
